Egor batch run failing with errors from rand.h

While running `egor batch r` I get the following error:

```
In file included from gen.cpp:9:
./rand.h:37:18: warning: declaration shadows a field of 'Rand' [-Wshadow]
    unsigned int seed = ReadSeedFromArgs(argc, argv);
                 ^
./rand.h:27:16: note: previous declaration is here
  unsigned int seed;
               ^
./rand.h:112:10: error: no member named 'random_shuffle' in namespace 'std'
    std::random_shuffle(charset.begin(), charset.end());
    ~~~~~^
1 warning and 1 error generated.
```

--- a/templates/resolver.go
+++ b/templates/resolver.go
@@ -141,6 +141,8 @@ class Rand {
   // an std::string of size 10 consisting only of lowercase letters and digits
   std::string String(size_t size, const int mask = LOWER | UPPER) {
     std::string charset;
+    std::random_device rd;
+    std::mt19937 g(rd());
 
     // Building the set of allowed charset from the mask
     if (mask & LOWER) {
@@ -153,7 +155,7 @@ class Rand {
       charset += DIGITS_SET;
     }
 
-    std::random_shuffle(charset.begin(), charset.end());
+    std::shuffle(charset.begin(), charset.end(), g);
 
     std::string res(size, ' ');
 

